db/db_impl/db_impl_open.cc

```diff
@@ -588,12 +588,13 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& log_numbers,
   bool stop_replay_for_corruption = false;
   bool flushed = false;
   uint64_t corrupted_log_number = kMaxSequenceNumber;
+  uint64_t min_log_number = MinLogNumberToKeep();
   for (auto log_number : log_numbers) {
-    if (log_number < versions_->min_log_number_to_keep_2pc()) {
```

Comment:
Isn't this a bug that is independent of `recycle_log_file_num`? When 2pc is disabled, `min_log_number_to_keep_2pc` is 0 which means all log files will be replayed, even if the manifest indicates a log file has already been fully processed. I'm not terribly familiar with the recovery code, so perhaps I'm missing something that increments `min_log_number_to_keep_2pc` when 2pc is disabled.

Reply:
I don't think there's a problem with replaying a WAL that has already been fully processed. Each operation will be processed and ignored after checking the manifest, which tells us the operation's column family has already been flushed past this WAL. So reading unnecessary old WALs would do a lot of no-ops, in my understanding.

Reply:
Ah, I was missing that `MemTableInserter` also checks that the recovering log number against the column family's log number.